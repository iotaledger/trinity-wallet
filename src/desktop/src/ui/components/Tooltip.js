import React from 'react';
import PropTypes from 'prop-types';
import Icon from 'ui/components/Icon';
import css from './tooltip.scss';

/**
 * Inline hover tooltip component
 */
export default class Tooltip extends React.PureComponent {
    static propTypes = {
        /** Tooltip text */
        tip: PropTypes.string.isRequired,
        /** Optional tooltip title */
        title: PropTypes.string,
    };

    render() {
        return (
            <div className={css.tooltip}>
                <Icon icon="info" size={21} />
                <div>
                    {this.props.title ? <strong>{this.props.title}</strong> : null}
                    <em>{this.props.tip}</em>
                </div>
            </div>
        );
    }
}
