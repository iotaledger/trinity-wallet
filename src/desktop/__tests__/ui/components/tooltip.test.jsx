import React from 'react';
import { shallow } from 'enzyme';

import Tooltip from 'ui/components/Tooltip';

const props = {
    tip: 'Lorem ipsum',
    title: 'Dolor sit amet',
};

describe('Tooltip component', () => {
    test('Render the component', () => {
        const wrapper = shallow(<Tooltip {...props} />);

        expect(wrapper).toMatchSnapshot();
    });

    test('Render with tip', () => {
        const wrapper = shallow(<Tooltip {...props} />);
        const titleEl = wrapper.find('em');

        expect(titleEl.text()).toEqual(props.tip);
    });

    test('Render with title', () => {
        const wrapper = shallow(<Tooltip {...props} />);
        const titleEl = wrapper.find('strong');

        expect(titleEl.text()).toEqual(props.title);
    });

    test('Render without title', () => {
        const mockProps = Object.assign({}, props, { title: null });
        const wrapper = shallow(<Tooltip {...mockProps} />);

        expect(wrapper.find('strong')).toHaveLength(0);
    });
});
