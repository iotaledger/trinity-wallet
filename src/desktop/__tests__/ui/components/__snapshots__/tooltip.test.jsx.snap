// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tooltip component Render the component 1`] = `
<div
  className="tooltip"
>
  <Icon
    icon="info"
    size={21}
  />
  <div>
    <strong>
      Dolor sit amet
    </strong>
    <em>
      Lorem ipsum
    </em>
  </div>
</div>
`;
